src/components/LocalFileUpload/custom-request-upload.ts

```diff
@@ -62,68 +55,94 @@ export default (
+                    {} as { [key: string]: string }
                 );
-            }
-        })
-        .catch((error) => {
-            let message = error.message;
-            if (error instanceof DOMException) {
-                message =
-                    "Please load a collection of single files that does not include a folder.";
-            }
-            setError({
-                level: error.level,
-                message: message,
-                htmlData: "",
+                const fileName = filesArr[simulariumFileIndex].name;
+
+                loadFunction({
+                    lastModified: filesArr[simulariumFileIndex].lastModified,
+                    name: fileName,
+                    data: simulariumFile,
+                    geoAssets: geoAssets,
+                });
+                // TS thinks onSuccess might be undefined
+                if (onSuccess) {
+                    onSuccess(
+                        {
+                            name: fileName,
+                            status: "done",
+                            url: "",
+                        },
+                        new XMLHttpRequest() // onSuccess needs an XMLHttpRequest arg
+                    );
+                }
+            })
+            .catch((error) => {
+                let message = error.message;
+                if (error instanceof DOMException) {
+                    message =
+                        "Please load a collection of single files that does not include a folder.";
+                }
+                setError({
+                    level: error.level,
+                    message: message,
+                    htmlData: "",
+                });
+                setViewerStatus({ status: VIEWER_ERROR });
+                clearSimulariumFile({ newFile: false });
+                clearUrlParams();
+                // TS thinks onError might be undefined
+                if (onError) {
+                    onError(error as UploadRequestError);
+                }
             });
-            setViewerStatus({ status: VIEWER_ERROR });
-            clearSimulariumFile({ newFile: false });
-            clearUrlParams();
-            // TS thinks onError might be undefined
-            if (onError) {
-                onError(error as UploadRequestError);
-            }
+    } catch (error) {
+        let message = error.message;
+        if (error instanceof DOMException) {
+            message =
+                "Please load a collection of single files that does not include a folder.";
+        }
+        setError({
+            level: error.level,
+            message: message,
+            htmlData: "",
         });
+        setViewerStatus({ status: VIEWER_ERROR });
+        clearSimulariumFile({ newFile: false });
+        clearUrlParams();
+        // TS thinks onError might be undefined
+        if (onError) {
+            onError(error as UploadRequestError);
+        }
+    }
```

Comment:
TODO I will try to do one more commit to DRY this error handling because there are two separate catch blocks here calling the same code.